packages/flutter/test/material/input_decorator_test.dart

```diff
@@ -248,6 +248,22 @@ void main() {
     expect(tester.getTopLeft(find.text('label')).dy, 20.0);
     expect(tester.getBottomLeft(find.text('label')).dy, 36.0);
     expect(getBorderColor(tester), Colors.transparent);
+
+    // labelAlignment: top positions the label at the text baseline.
+    await tester.pumpWidget(
+      buildInputDecorator(
+        isEmpty: true,
+        isFocused: false,
+        decoration: const InputDecoration(
+          labelText: 'label',
+          labelAlignment: TextFieldLabelAlignment.top,
+        ),
+      ),
+    );
+    await tester.pumpAndSettle();
+    expect(tester.getSize(find.byType(InputDecorator)), const Size(800.0, 56.0));
```

Comment:
Also test this against a multiline text field (the default child for buildInputDecorator is `Text('text')`.

Rather than checking absolute top/left values, verify that the label is in the same location as the hint (other tests verify that the hint ends up in the right place).